internal/image/validate.go

```diff
@@ -141,6 +142,48 @@ func ValidateImage(ctx context.Context, comp app.SnapshotComponent, snap *app.Sn
 	return out, nil
 }
 
+// ValidateImageWithVSACheck executes validation with VSA expiration checking.
+// If a valid, unexpired VSA exists in Rekor, validation is skipped.
+func ValidateImageWithVSACheck(ctx context.Context, comp app.SnapshotComponent, snap *app.SnapshotSpec, p policy.Policy, evaluators []evaluator.Evaluator, detailed bool, vsaExpiration time.Duration, rekorURL string) (*output.Output, error) {
```

Comment:
This function is tightly coupled with Rekor. You might be able to use the `VSARetriever` interface and pass in the appropriate retriever.